jaxrs/src/main/java/io/airlift/jaxrs/tracing/TracingServletFilter.java

```diff
@@ -0,0 +1,33 @@
+package io.airlift.jaxrs.tracing;
+
+import io.opentelemetry.context.Scope;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import java.io.IOException;
+
+public final class TracingServletFilter
+        implements Filter
+{
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException
+    {
+        request.setAttribute(TracingFilter.REMOTE_ADDRESS, request.getRemoteAddr());
+
+        try {
+            chain.doFilter(request, response);
+        }
+        catch (Throwable t) {
+            try (var ignored = (Scope) request.getAttribute(TracingFilter.REQUEST_SCOPE)) {
```

Comment:
Add a comment explaining why this is not just a finally block (avoid exception getting overwritten)